PhysicsTools/PatAlgos/python/tools/jetCollectionTools.py

```diff
@@ -0,0 +1,512 @@
+    tagName  = recoJetInfo.jetTagName
+
+    if inputCollection == "slimmedJets":
+      assert(jetLower == "ak4pfchs")
+    elif inputCollection == "slimmedJetsAK8":
+      assert(jetLower == "ak8pfpuppi")
+    elif inputCollection == "slimmedJetsPuppi":
+      assert(jetLower == "ak4pfpuppi")
+    elif inputCollection == "slimmedCaloJets":
+      assert(jetLower == "ak4calo")
+    
+    #=======================================================
+    #
+    # If jet collection in MiniAOD is not 
+    # specified, build the jet collection.
+    #
+    #========================================================
+    if not inputCollection or recoJetInfo.doCalo:
+      print("jetCollectionTools::RecoJetAdder::addRecoJetCollection: inputCollection not specified. Building recojet collection now")
+
+      #=======================================================
+      #
+      # Prepare the inputs to jet clustering
+      #
+      #========================================================
+      #
+      # Set up PF candidates
+      #
+      pfCand = self.pfLabel
+      #
+      # Setup PU method for PF candidates
+      # 
+      if recoJetInfo.jetPUMethod not in [ "", "cs" ]:
+        pfCand += recoJetInfo.jetPUMethod
+      #
+      #
+      #
+      if pfCand not in self.prerequisites:
+        #
+        # Skip if no PU Method or CS specified
+        #
+        if recoJetInfo.jetPUMethod in [ "", "cs" ]:
+          pass
+        #
+        # CHS
+        #
+        elif recoJetInfo.jetPUMethod == "chs":
+          setattr(proc, pfCand,
+            cms.EDFilter("CandPtrSelector",
+              src = cms.InputTag(self.pfLabel),
+              cut = cms.string("fromPV"),
+            )
+          )
+          self.prerequisites.append(pfCand)
+        #
+        # PUPPI
+        #
+        elif recoJetInfo.jetPUMethod == "puppi":
+          setattr(proc, pfCand,
+            puppi.clone(
+              candName   = self.pfLabel,
+              vertexName = self.pvLabel,
+            )
+          )
+          self.prerequisites.append(pfCand)
+        #
+        # Softkiller
+        #
+        elif recoJetInfo.jetPUMethod == "sk":
+          setattr(proc, pfCand,
+            softKiller.clone(
+              PFCandidates = self.pfLabel,
+              rParam       = recoJetInfo.jetSizeNr,
+            )
+          )
+          self.prerequisites.append(pfCand)
+        else:
+          raise RuntimeError("Currently unsupported PU method: '%s'" % recoJetInfo.jetPUMethod)
+      
+      #============================================
+      #
+      # Create the recojet collection
+      #
+      #============================================
+      if not recoJetInfo.doCalo:
+        jetCollection = '{}Collection'.format(tagName)
+
+        if jetCollection in self.main:
+          raise ValueError("Step '%s' already implemented" % jetCollection)
+
+        setattr(proc, jetCollection,
+          cms.EDProducer("FastjetJetProducer",
```

Comment:
isn't there an existing reference jet producer to clone from instead of rebuilding that config from scratch?
e.g. ak4PFJets.
OTOH, it looks like only ```AnomalousCellParameters``` are not replaced/modified here. So, perhaps OK.

Reply:
I can clone [ak4PFJetsCS](https://github.com/cms-sw/cmssw/blob/CMSSW_11_0_0_pre6/RecoJets/JetProducers/python/ak4PFJets_cfi.py#L28) from RecoJets/JetProducers/python/ak4PFJets_cfi.py and use it here.